src/snowflake/snowpark/stored_procedure.py

```diff
@@ -102,8 +103,13 @@ def __call__(
 
         if self._anonymous_sp_sql:
             call_sql = generate_call_python_sp_sql(session, self.name, *args)
-            df = session.sql(f"{self._anonymous_sp_sql}{call_sql}")
+            query = f"{self._anonymous_sp_sql}{call_sql}"
+            df = session.sql(query)
             if self._is_return_table:
+                qid = session._conn.execute_and_get_sfqid(
```

Comment:
Does the same query get run twice?

Reply:
`df = session.sql(query)` does not run the query, doing an internal collect or running function `execute_and_get_sfqid` does. I see df on line 107 is only used for the final return on line 114. I'll update the PR and move it so it become less confusing.

Reply:
I understand now. Thanks for explaining.